src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -233,14 +233,15 @@ doLookupBlock dbenv (bheight, bhash) = runBlockEnv dbenv $ do
             \ AND hash = ?;"
 
 doGetBlockParent :: Db -> (BlockHeight, BlockHash) -> IO (Maybe BlockHash)
-doGetBlockParent dbenv (bh, hash) =
-    if bh == 0 then return Nothing else
-      doLookupBlock dbenv (bh, hash) >>= \blockFound ->
-        if not blockFound then return Nothing else runBlockEnv dbenv $ do
-          r <- callDb "getBlockParent" $ \db -> qry db qtext [SInt (fromIntegral (pred bh))] [RBlob]
-          case r of
-            [[SBlob blob]] -> either (internalError . T.pack) (return . return) $! Data.Serialize.decode blob
-            _ -> internalError "doGetBlockParent: output mismatch"
+doGetBlockParent dbenv (bh, hash) = do
+    blockFound <- doLookupBlock dbenv (bh, hash)
```

Comment:
Actually, not sure. I think, I hoped that for genesis blocks `doLookupBlock` would return `not blockFound`.

Should, we call `error` instead? Well, it would probably result in `internalError` in any case. So, far this hasn't triggered.

Reply:
The problem is that within the checkpointer we don't have the chainweb version and/or the chain id available, so we don't know what the genesis height is.

Since the checkpointer is just and function dictionary, with no explicit state, there's no place where we could store this kind of information.